Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -201,6 +202,302 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
 #endif
         }
         );
+#if defined(WARPX_USE_GPUCLOCK)
+    if( load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::GpuClock) {
```

Comment:
Source of many bugs in the past:
```suggestion
    if (cost && load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::GpuClock) {
```

This function is sometimes called early, before cost arrays are allocated.

Please also add in line 93 above to avoid a memory leak of `cost_real`.